crypto/src/main/kotlin/web5/sdk/crypto/Ed25519.kt

```diff
@@ -0,0 +1,163 @@
+ * TODO: Insert example usage here.
+ *
+ * @property algorithm Specifies the JWS algorithm type. For Ed25519, this is `EdDSA`.
+ * @property keyType Specifies the key type. For Ed25519, this is `OKP`.
+ * @property pubMulticodec A byte array representing the multicodec prefix for an Ed25519 public key.
+ * @property privMultiCodec A byte array representing the multicodec prefix for an Ed25519 private key.
+ */
+
+public object Ed25519 : KeyGenerator, Signer {
+  override val algorithm: Algorithm = JWSAlgorithm.EdDSA
+  override val keyType: KeyType = KeyType.OKP
+
+  public const val pubMulticodec: Int = 0xed
+  public const val privMultiCodec: Int = 0x1300
+
+  /**
+   * Generates a private key utilizing the Ed25519 algorithm.
+   *
+   * @param options (Optional) Additional options to control the key generation process.
+   * @return The generated private key in JWK format.
+   */
+  override fun generatePrivateKey(options: KeyGenOptions?): JWK {
+    return OctetKeyPairGenerator(Curve.Ed25519)
+      .algorithm(JWSAlgorithm.EdDSA)
+      .keyIDFromThumbprint(true)
+      .keyUse(KeyUse.SIGNATURE)
+      .generate()
+      .toOctetKeyPair()
+  }
+
+  /**
+   * Derives the public key corresponding to a given private key.
+   *
+   * @param privateKey The private key in JWK format.
+   * @return The corresponding public key in JWK format.
+   */
+  override fun getPublicKey(privateKey: JWK): JWK {
+    require(privateKey is OctetKeyPair) { "private key must be an Octet Key Pair (kty: OKP)" }
+
+    return privateKey.toOctetKeyPair().toPublicJWK()
+  }
+
+  override fun privateKeyToBytes(privateKey: JWK): ByteArray {
+    validateKey(privateKey)
+
+    return privateKey.toOctetKeyPair().decodedD
+  }
+
+  override fun publicKeyToBytes(publicKey: JWK): ByteArray {
+    validateKey(publicKey)
+
+    return publicKey.toOctetKeyPair().decodedX
+  }
+
+  override fun bytesToPrivateKey(privateKeyBytes: ByteArray): JWK {
+    val privateKeyParameters = Ed25519PrivateKeyParameters(privateKeyBytes, 0)
+    val publicKeyBytes = privateKeyParameters.generatePublicKey().encoded
+
+    val base64UrlEncodedPrivateKey = Convert(privateKeyBytes).toBase64Url(padding = false)
+    val base64UrlEncodedPublicKey = Convert(publicKeyBytes).toBase64Url(padding = false)
+
+    return OctetKeyPair.Builder(Curve.Ed25519, Base64URL(base64UrlEncodedPublicKey))
+      .algorithm(algorithm)
+      .keyIDFromThumbprint()
+      .d(Base64URL(base64UrlEncodedPrivateKey))
+      .keyUse(KeyUse.SIGNATURE)
+      .build()
+  }
+
+  override fun bytesToPublicKey(publicKeyBytes: ByteArray): JWK {
+    val base64UrlEncodedPublicKey = Convert(publicKeyBytes).toBase64Url(padding = false)
+
+    return OctetKeyPair.Builder(Curve.Ed25519, Base64URL(base64UrlEncodedPublicKey))
+      .algorithm(algorithm)
+      .keyIDFromThumbprint()
+      .keyUse(KeyUse.SIGNATURE)
+      .build()
+  }
+
+  override fun sign(privateKey: JWK, payload: Payload, options: SignOptions?): String {
```

Comment:
maybe "Returns a Compact JWS" instead of  "Returns a JWS Signature ..." JWS Signature is redundant

Reply:
Oh I though you were returning the "JWS Signature" as defined in https://www.rfc-editor.org/rfc/rfc7515#section-2

Would it make sense to return a `JWSObject` then?

Reply:
yeah i'm going back and forth right now about what `sign` here should return. I think we should merge as is so that we can get this into mainline and adjust as needed in order to support all of the signing needs we have: e.g. jwt, jws, plain ole signatures as byte arrays etc.